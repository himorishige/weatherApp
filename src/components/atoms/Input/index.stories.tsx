import React from 'react';
import { Meta, Story } from '@storybook/react';
import Input, { Props } from '.';

export default {
  title: 'WeatherApp/Atoms/Input',
  component: Input,
} as Meta;

const Template: Story<Props> = (args) => <Input {...args} />;

export const SearchInput = Template.bind({});
SearchInput.args = {
  disabled: false,
};
