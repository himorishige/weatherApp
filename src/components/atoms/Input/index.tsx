/** @jsxImportSource @emotion/react */
import React from 'react';
import { css } from '@emotion/react';

export type Props = {
  disabled?: boolean;
  place?: string;
  inputHandler?: (event: React.ChangeEvent<HTMLInputElement>) => void;
};
const Input: React.VFC<Props> = ({ place, disabled = false, inputHandler }) => {
  return (
    <>
      <input
        type="text"
        placeholder="（例）東京"
        aria-label="search input"
        value={place}
        defaultValue=""
        data-testid="input"
        css={inputStyle}
        disabled={disabled}
        onChange={inputHandler}
      />
    </>
  );
};

export default Input;

const inputStyle = css`
  padding: 0.5rem;
  width: 100%;
  border-radius: 4px;
  border: 1px solid #999999;
  &::placeholder {
    color: #ccc;
  }
  &:disabled {
    color: #ccc;
    background-color: #f8f8f8;
    cursor: not-allowed;
  }
`;
