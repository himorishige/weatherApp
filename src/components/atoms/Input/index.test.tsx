import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Input from '.';

describe('Atoms/Input', () => {
  it('placeholderに（例）東京が表示される', () => {
    render(<Input />);
    const inputElement = screen.getByTestId('input') as HTMLInputElement;
    expect(inputElement.placeholder).toBe('（例）東京');
  });
  it('検索フィールドに1を入力したら1が表示される', () => {
    render(<Input />);
    const inputElement = screen.getByTestId('input') as HTMLInputElement;
    fireEvent.change(inputElement, { target: { value: '1' } });
    expect(inputElement.value).toBe('1');
  });
  it('正しくレンダリングされている', () => {
    const inputField = renderer.create(<Input />).toJSON();
    expect(inputField).toMatchSnapshot();
  });
});
