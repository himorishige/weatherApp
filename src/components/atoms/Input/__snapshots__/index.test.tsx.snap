// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/Input 正しくレンダリングされている 1`] = `
<input
  aria-label="search input"
  className="css-31czxj-Input"
  data-testid="input"
  defaultValue=""
  disabled={false}
  placeholder="（例）東京"
  type="text"
/>
`;
